docs/DeveloperGuide.md

```diff
@@ -1,34 +1,833 @@
+* The `Show` review is then set to "This show is great", as per the user input.
+
+**Step 5**
+
+* The `Show` is updated back into the `ShowList`.
+
+**Step 6**
+
+* The changes are reflected back to the user. Concurrently, the changes will be saved to the showList.txt file.
+
+### DeleteReviewCommand
+
+The `deletereview` command takes in 1 parameter, the show which review is to be deleted.
+The command is then invoked by the `InputParser` method `parseDeleteReview`.
+
+Given below is an example usage scenario and how the `DeleteCommand` Class behaves at each step.
+
+**Step 1**
+
+* The user types in `deletereview friends` , assuming that friends has been added by the user beforehand.
+The `parseInput` method in `InputParser` class is called to parse the command.
+
+>**[NOTE]** 
+>
+> Customised `NullPointerException` will be thrown when show entered by user is not found in the show list.
+
+**Step 2**
+
+* A new instance of `DeleteReviewCommand` class is called and the command is returned to the main program. 
+The `deleteReview` method in `DeleteReviewCommand` class is called.
+
+**Step 3**
+
+* The `deleteReview` method starts with retrieving the `Show` from the `ShowList`.
+
+**Step 4**
+
+* The `Show` review is then set to "null", essentially deleting it.
+
+**Step 5**
+
+* The `Show` is updated back into the `ShowList`.
+
+**Step 6**
+
+The changes are reflected back to the user. Concurrently, the changes will be saved to the showList.txt file.
+
+### WatchCommand
+
+The `WatchCommand` class extends `Command` by providing methods to 
+increment the current episode in the persistent watch history of the user. It also updates the watch time limit as indicated previously by the user.
+
+Given below is an example usage scenario and how the `WatchCommand` class behaves at each step.
+
+**Step 1**
+
+* The user types in `watch friends` , assuming that friends has been added by the user beforehand.
+The `parseInput` method in `InputParser` class is called to parse the command.
+
+> **[NOTE]** Customised `IndexOutOfBoundsException` and `NullPointerException` will be thrown if the user enters invalid commands.
+
+**Step 2**
+
+* A new instance of `WatchCommand` class is called and the command is returned to the main program. 
+The `processCommand` method in `WatchCommand` class is called.
+
+**Step 3**
+
+* The `processCommand` method in `WatchCommand` class is then called. This method does three main things:
+
+1.Check the status of user's watch progress: In middle of series , finished season and finished series.
+
+2.Increment current episode  and new season if applicable. No change is done if user has finished series. 
+
+3.Reflect the new changes to the user. A prompt is made to the user if the user has already finished the series. Changes are also saved in the `userData.txt` file.
+
+
+The following sequence diagram summarises what happens when a user executes a `WatchCommand`:
+
+ <img src = "images/WatchCommandSequence.png" width = "700">
```

Comment:
![image](https://user-images.githubusercontent.com/57038559/97527203-af0cf800-19e5-11eb-9dd3-4baf45e59195.png)


Some suggestions:
- for the part circled in red, I think you're missing a return arrow?
- for the part circled in green, I think there should be an activation bar there.